vertx-mysql-client/src/main/java/io/vertx/mysqlclient/MySQLChangeUserOptions.java

```diff
@@ -0,0 +1,249 @@
+   * Specify the default database for the re-authentication.
+   *
+   * @param database the database name to specify
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setDatabase(String database) {
+    checkParameterNonNull(database, "Database name can not be null");
+    this.database = database;
+    return this;
+  }
+
+  /**
+   * @return the value of current connection attributes
+   */
+  public Map<String, String> getProperties() {
+    return properties;
+  }
+
+  /**
+   * Set connection attributes which will be sent to server at the re-authentication.
+   *
+   * @param properties the value of properties to specify
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setProperties(Map<String, String> properties) {
+    checkParameterNonNull(properties, "Properties can not be null");
+    this.properties = properties;
+    return this;
+  }
+
+  /**
+   * Add a property for this client, which will be sent to server at the re-authentication.
+   *
+   * @param key   the value of property key
+   * @param value the value of property value
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions addProperty(String key, String value) {
+    checkParameterNonNull(key, "Property key can not be null");
+    checkParameterNonNull(value, "Property value can not be null");
+    this.properties.put(key, value);
+    return this;
+  }
+
+  /**
+   * Get the collation for the connection.
+   *
+   * @return the MySQL collation
+   */
+  public String getCollation() {
+    return collation;
+  }
+
+  /**
+   * Set the collation for the connection.
+   *
+   * @param collation the collation to set
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setCollation(String collation) {
+    if (collation != null && !MySQLCollation.SUPPORTED_COLLATION_NAMES.contains(collation)) {
+      throw new IllegalArgumentException("Unsupported collation: " + collation);
+    }
+    this.collation = collation;
+    return this;
+  }
+
+  /**
+   * Get the charset for the connection.
+   *
+   * @return the MySQL collation
+   */
+  public String getCharset() {
+    return charset;
+  }
+
+  /**
+   * Set the charset for the connection.
+   *
+   * @param charset the charset to set
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setCharset(String charset) {
+    if (charset != null && !MySQLCollation.SUPPORTED_CHARSET_NAMES.contains(charset)) {
+      throw new IllegalArgumentException("Unsupported charset: " + charset);
+    }
+    this.charset = charset;
+    return this;
+  }
+
+  /**
+   * Get the path of the server RSA public key.
+   *
+   * @return a reference to this, so the API can be used fluently
+   */
+  public String getServerRsaPublicKeyPath() {
+    return serverRsaPublicKeyPath;
```

Comment:
do we need to have the server prefix ? because this key has to be for the server right ?

Reply:
it's better naming because this is the server public key which is stored locally on the client side.